src/eviction/lru.c

```diff
@@ -163,48 +163,121 @@ void evp_lru_init_cline(ocf_cache_t cache, ocf_cache_line_t cline)
 	node->next = end_marker;
 }
 
+static inline struct ocf_lru_list *evp_get_list(ocf_cache_t cache,
+		ocf_cache_line_t cline)
+{
+	ocf_part_id_t part_id = ocf_metadata_get_partition_id(cache, cline);
+	struct ocf_user_part *part = &cache->user_parts[part_id];
+	uint32_t ev_list = (cline % OCF_NUM_EVICTION_LISTS);
+
+	return metadata_test_dirty(cache, cline) ?
+		&part->runtime->eviction[ev_list].policy.lru.dirty :
+		&part->runtime->eviction[ev_list].policy.lru.clean;
+}
 
 /* the caller must hold the metadata lock */
 void evp_lru_rm_cline(ocf_cache_t cache, ocf_cache_line_t cline)
 {
-	ocf_part_id_t part_id = ocf_metadata_get_partition_id(cache, cline);
-	struct ocf_user_part *part = &cache->user_parts[part_id];
 	struct ocf_lru_list *list;
 
-	list = metadata_test_dirty(cache, cline) ?
-		&part->runtime->eviction.policy.lru.dirty :
-		&part->runtime->eviction.policy.lru.clean;
-
+	list = evp_get_list(cache, cline);
 	remove_lru_list(cache, list, cline);
 }
 
+static inline void lru_iter_init(ocf_cache_t cache, ocf_part_id_t part_id,
+	uint32_t start_evp, bool clean, struct ocf_lru_iter *iter)
```

Comment:
As a convention constructor takes pointer to initialized object as a first parameter.